src/android/GeofenceTransitionsIntentService.java

```diff
@@ -0,0 +1,265 @@
+ * limitations under the License.
+ */
+
+package com.example.scandevice;
+
+import android.app.IntentService;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.BitmapFactory;
+import android.graphics.Color;
+import android.location.Criteria;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.app.TaskStackBuilder;
+import android.text.TextUtils;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.google.android.gms.location.Geofence;
+import com.google.android.gms.location.GeofencingEvent;
+
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Listener for geofence transition changes.
+ *
+ * Receives geofence transition events from Location Services in the form of an Intent containing
+ * the transition type and geofence id(s) that triggered the transition. Creates a notification
+ * as the output.
+ */
+public class GeofenceTransitionsIntentService extends IntentService implements LocationListener {
+
+    protected static final String TAG = "GeofenceTransitionsIS";
+    private double lat, lon, alt;
+    private String timestamp = "yyyyMMdd'T'HH:mm:ss";
+
+    public DBHelper GPS_dbManager;
+    public Geofence geofence;
+
+
+
+
+    String provider;
+
+    /**
+     * This constructor is required, and calls the super IntentService(String)
+     * constructor with the name for a worker thread.
+     */
+    public GeofenceTransitionsIntentService() {
+        // Use the TAG to name the worker thread.
+        super(TAG);
+}
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        LocationManager locationManager = (LocationManager)getSystemService(Context.LOCATION_SERVICE);
+
+        Criteria criteria = new Criteria();
+
+        // Getting the name of the provider that meets the criteria
+        provider = locationManager.getBestProvider(criteria, true);
+
+        if(provider == null && !locationManager.isProviderEnabled(provider)){
+            // Get the location from the given provider
+            List<String> list = locationManager.getAllProviders();
+
+            for(int i = 0; i < list.size(); i++){
+                //Get device name;
+                String temp = list.get(i);
+                //check usable
+                if(locationManager.isProviderEnabled(temp)){
+                    provider = temp;
+                    break;
+                }
+            }
+        }
+        //get location where reference last.
+        Location location = locationManager.getLastKnownLocation(provider);
+
+        if(location == null)
+            Toast.makeText(this, "There are no available position information providers.", Toast.LENGTH_SHORT).show();
```

Comment:
never call the UI from a plugin. instead, use a JavaScript callback so that the app developer can react to this.